### PR TITLE
Skips to ask specifications for optional workspace

### PR DESCRIPTION
**Pull request by pratap0007**
# Changes
This patch skips to ask specifications for optional workspace in an interactive mode  if workspace of t/p/ct is optional and optional workspace skip flag(--skip-optional-workspace)is passed  with `tkn t/p/ct start` command 

Closes: #1444 

Signed-off-by: Shiv Verma <shverma@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

### CODE REVIEW

File: pkg/cmd/clustertask/start.go [hunk 1]
```diff
@@ -195,6 +196,7 @@ For passing the workspaces via flags:
 	c.Flags().StringVarP(&opt.PrefixName, "prefix-name", "", "", "specify a prefix for the TaskRun name (must be lowercase alphanumeric characters)")
 	c.Flags().StringVar(&opt.PodTemplate, "pod-template", "", "local or remote file containing a PodTemplate definition")
 	c.Flags().BoolVar(&opt.UseParamDefaults, "use-param-defaults", false, "use default parameter values without prompting for input")
+	c.Flags().BoolVarP(&opt.SkipOptionalWorkspace, "skip-optional-workspace", "", false, "skip to ask specification for optional workspace")
```
Comment (vinamra28) [inline pkg/cmd/clustertask/start.go:199]: how about adding the description -> `skips the prompt for optional workspaces` ?

File: pkg/cmd/task/start_test.go [hunk 2]
```diff
@@ -5684,3 +5684,59 @@ func TestTaskStart_ExecuteCommand_v1beta1(t *testing.T) {
...
+
+func Test_start_task_with_skip_optional_workspace_flag(t *testing.T) {
+	tasks := []*v1alpha1.Task{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "task-1",
+				Namespace: "ns",
+			},
+			Spec: v1alpha1.TaskSpec{
+				TaskSpec: v1beta1.TaskSpec{
```
Comment (vinamra28) [inline pkg/cmd/task/start_test.go:5696]: instead of using `v1alpha1` structs, can we use `v1beta1` structs here?

File: pkg/cmd/task/start.go [hunk 3]
```diff
@@ -218,7 +219,7 @@ For passing the workspaces via flags:
 	c.Flags().StringVarP(&opt.PrefixName, "prefix-name", "", "", "specify a prefix for the TaskRun name (must be lowercase alphanumeric characters)")
 	c.Flags().BoolVarP(&opt.UseParamDefaults, "use-param-defaults", "", false, "use default parameter values without prompting for input")
 	c.Flags().StringVar(&opt.PodTemplate, "pod-template", "", "local or remote file containing a PodTemplate definition")
-
+	c.Flags().BoolVarP(&opt.SkipOptionalWorkspace, "skip-optional-workspace", "", false, "skip to ask specification for optional workspace")
```
Comment (vinamra28) [inline pkg/cmd/task/start.go:222]: same as previous

File: pkg/cmd/pipeline/start.go [hunk 4]
```diff
@@ -191,6 +192,7 @@ For passing the workspaces via flags:
 	c.Flags().StringVarP(&opt.Filename, "filename", "f", "", "local or remote file name containing a Pipeline definition to start a PipelineRun")
 	c.Flags().BoolVarP(&opt.UseParamDefaults, "use-param-defaults", "", false, "use default parameter values without prompting for input")
 	c.Flags().StringVar(&opt.PodTemplate, "pod-template", "", "local or remote file containing a PodTemplate definition")
+	c.Flags().BoolVarP(&opt.SkipOptionalWorkspace, "skip-optional-workspace", "", false, "skip to ask specification for optional workspace")
```
Comment (vinamra28) [inline pkg/cmd/pipeline/start.go:195]: same as above

**Review by vinamra28 — COMMENTED** [review]
/lgtm

**Review by vdemeester — APPROVED** [review]

**Pull request merged**